Hoxtygen/next-fullstack
# refactor: rewrite signup API to use Java API

## What does this PR do?
Refactored the signup API  to use endpoints from the Java Spring boot

## Description of Task to be completed
* Delete all files related to Prisma and its functionalities.
* Delete Nodejs endpoints. 
* Delete react context
* Delete authentication reducer
* Add custom hook for connecting to signup endpoint
* Refactored the `SignupForm` component to use the custom hook for signup

## How should this be manually tested?
* Clone the app
* Open a terminal on your PC and run `git clone <repo_url>`
* `cd` into the project directory and run `yarn install`.
* After package installation run `yarn dev`
* Open a tab on your browser and navigate to `http://localhost:3000/signup`
* Open the network tab, and fill out the signup form.
* Click on the submit button to submit the form
* Check the `Headers` in the network tab, should point. to `http://localhost:8080/api/v1/auth/register`

## What are the relevant Jira board stories?
[HOM-2](https://wasiu-dev.atlassian.net/jira/software/projects/HOM/boards/2?selectedIssue=HOM-2)

[HOM-2]: https://wasiu-dev.atlassian.net/browse/HOM-2?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
